compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1656,17 +1668,19 @@ public BType transform(BLangUserDefinedType userDefinedTypeNode, AnalyzerData da
                     userDefinedTypeNode.symbol = tSymbol;
                     return tSymbol.type;
                 }
-            } else if (Symbols.isTagOn(tempSymbol, SymTag.VARIABLE) && env.node.getKind() == NodeKind.FUNCTION_TYPE) {
+            } else if (Symbols.isTagOn(tempSymbol, SymTag.VARIABLE) && (env.node.getKind() == NodeKind.FUNCTION_TYPE
+                    || env.node.getKind() == NodeKind.TUPLE_TYPE_NODE)) {
```

Comment:
Why do we need this addition for tuples?

Reply:
Any update on this?

Reply:
IIR this was added since after adding to `BVarSymbol`s for tuple members in cases like `function getTuple(typedesc<int|string> td1, typedesc<record {}> td2, typedesc<float|boolean> td3 = float) returns [td1, td2, td3]`, `td1` etc in `[td1, td2, td3]` were undefined. Still checking if there is other use cases.